src/Moryx.Communication.Mqtt.Endpoints/ResourceRpcEndpoint.cs

```diff
@@ -0,0 +1,99 @@
+            messageBuilder.WithTopic(context.RequestMessage.ResponseTopic);
+
+            var identifier = context.FromParameterValues<string>("identifier");
+            var methodName = context.FromParameterValues<string>("methodName") ?? string.Empty;
+            var resource = resourceManagement.GetResource<IResource>(x =>
+                x is IIdentifiableObject obj && obj.Identity.Identifier == identifier
+                || x.Name == identifier);
+            return FindAndInvokeMethod(context, messageBuilder, identifier, methodName, resource);
+        });
+    }
+
+    private MqttApplicationMessage? FindAndInvokeMethod(
+        MqttEndpointContext context,
+        MqttApplicationMessageBuilder responseBuilder,
+        string? identifier,
+        string methodName,
+        IResource? resource)
+    {
+        var methodToInvoke = resource
+                        ?.GetType()
+                        .GetInterfaces()
+                        .Select(i => i.GetMethod(methodName))
+                        .FirstOrDefault(x => x is not null);
+
+        if (resource is null || methodToInvoke is null)
+        {
+            logger.Log(LogLevel.Error, "No resource or method found with the given Identifier:`{id}`, method:`{method}`.", identifier, methodName);
+            return null;
+        }
+
+        var methodToInvokeParameters = methodToInvoke.GetParameters();
+        var parameters = new object?[methodToInvokeParameters.Length];
+
+        for (var i = 0; i < methodToInvokeParameters.Length; i++)
+        {
+            try
+            {
+                // get the values of the method's parameters from the body of the message
+                parameters[i] = context.FromBody(methodToInvokeParameters[i].Name!, methodToInvokeParameters[i].ParameterType);
+            }
+            catch (Exception ex)
+            {
+                logger.LogError(ex,
+                    "Impossible to get Method '{method}' parameter '{parameter}' from the Payload.",
+                    methodName,
+                    methodToInvokeParameters[i].Name);
+                continue;
+            }
+        }
+        var methodResult = methodToInvoke.Invoke(resource, parameters);
+        if (methodResult?.GetType() == typeof(void))
+        {
+            responseBuilder.WithPayload(MqttMessageSerialization.GetJsonPayload(new { }, options.JsonSerializerOptions));
+            return responseBuilder.Build();
+        }
+
+        // TODO: what if the method is a Task or Task<T> ?
+        var topic = string.IsNullOrEmpty(context.RequestMessage.ResponseTopic)
+                        ? context.RequestMessage.Topic.Replace("invoke", "invoked")
+                        : context.RequestMessage.ResponseTopic;
+        responseBuilder.WithTopic(topic);
+        var payload = methodResult ?? new { };
+        var jsonPayload = MqttMessageSerialization.GetJsonPayload(new MethodInvocationResult
```

Comment:
We should document somewhere that this serialization is used and how it works for the `ResourceRpcEndpoint` 😊